src/backwardproblem.cpp

```diff
@@ -131,12 +131,12 @@ void BackwardProblem::handleEventB() {
                                           xdot_disc,
                                           xdot_old_disc);
 
-        for (int ix = 0; ix < model->nxtrue_solver; ++ix) {
-            for (int iJ = 0; iJ < model->nJ; ++iJ) {
-                if (model->nz > 0) {
+        if (model->nz > 0) {
+            for (int ix = 0; ix < model->nxtrue_solver; ++ix) {
+                for (int iJ = 0; iJ < model->nJ; ++iJ) {
```

Comment:
better vectorization if we swap around loops over ix and jJ?

Reply:
For rhs better as is, for lhs better if swapped. Improvement unclear, I'd say. But anyhow not vectorizable because loop body can throw.